contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -566,6 +559,28 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         return _getNextMessageID(chainID);
     }
 
+    /**
+     * @dev Returns the number of receipts that have been received from the given chain ID.
+     */
+    function getReceiptQueueSize(bytes32 chainID)
+        external
+        view
+        returns (uint256)
+    {
+        return outstandingReceipts[chainID].size();
+    }
+
+    /**
+     * @dev Returns the receipt at the given index in the queue for the given chain ID.
+     */
+    function getReceiptAtIndex(bytes32 chainID, uint256 index)
```

Comment:
Same as above, and just add requirement note for index

Reply:
fixed